fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,421 @@
+"""
+Finds leaks between splits.
+"""
+
+from copy import copy
+import warnings
+
+from fiftyone import ViewField as F
+import fiftyone.core.brain as fob
+import fiftyone.brain.similarity as sim
+import fiftyone.core.utils as fou
+import fiftyone.core.validation as fov
+
+fbu = fou.lazy_import("fiftyone.brain.internal.core.utils")
+
+_DEFAULT_MODEL = "resnet18-imagenet-torch"
+
+
+def compute_leaky_splits(
+    samples,
+    brain_key=None,
+    split_views=None,
+    split_field=None,
+    split_tags=None,
+    threshold=0.2,
+    similarity_brain_key=None,
+    embeddings_field=None,
+    model=None,
+    model_kwargs=None,
+    similarity_backend=None,
+    similarity_config_dict=None,
+    **kwargs,
+):
+    """See ``fiftyone/brain/__init__.py``."""
+
+    fov.validate_collection(samples)
+
+    config = LeakySplitsConfig(
+        split_views,
+        split_field,
+        split_tags,
+        similarity_brain_key,
+        embeddings_field,
+        model,
+        model_kwargs,
+        similarity_backend,
+        similarity_config_dict,
+        **kwargs,
+    )
+    brain_method = config.build()
+    brain_method.ensure_requirements()
+
+    if brain_key is not None:
+        # Don't allow overwriting an existing run with same key, since we
+        # need the existing run in order to perform workflows like
+        # automatically cleaning up the backend's index
+        brain_method.register_run(samples, brain_key, overwrite=False)
```

Comment:
You should check to make sure this doesn't break anything, but by adding `cleanup=False` in `register_run()` and commenting out the `cleanup()` method in `LeakySplits` below, I was able to get rid of the error with `Limit` not being registered

Reply:
This works for registration and shouldn't break anything as far as I can tell but it doesn't fix the core issue when deleting a run.

Reply:
Really? It did for me, at least when running `dataset.delete_brain_runs()`